---
title: 'LOAD DATA'

category: User Interface
layout: null
type: ui
---
Clicking this button will expand a panel for loading datasets provided by Cellar or datasets uploaded by the user.
<br>
![CODEX Main Plot](images/codex-load-dataset.png)
<br>


* **<span class='mbutton'>UPLOAD</span>**
    * This button is for uploading your own dataset. We accept feature barcode sequences in
    [cellranger](https://support.10xgenomics.com/single-cell-gene-expression/software/pipelines/latest/output/matrices) or [spaceranger](https://support.10xgenomics.com/spatial-gene-expression/software/pipelines/latest/output/matrices)
    format as a <span class='extension'>gz</span> file,
    or alternatively <span class='extension'>csv</span> or <span class='extension'>h5ad</span> data formats. 
    <br>
    <span class="warn">!!<span class="tooltip">Attention</span></span> In case of a <span class='extension'>csv</span> or <span class='extension'>h5ad</span> file,
    make sure the data follows the cell $$\small\times $$ gene format,
    i.e., rows of the matrix correspond to cells and columns to genes. Additionaly, row and columns names must be provided.
        * Example cell barcode: <span class='keyword'>TTACCATTCCAGCACG</span>. Example gene name (can be in ensembl format or HGNC symbol):
        <span class='keyword'>ENSG00000211699</span> or <span class='keyword'>TRGV3</span>.

    * <span class="good">GP<span class="tooltip">Good Practice</span></span><span class="pn">PN<span class="tooltip">Note to Programmers</span></span>
    We recommend uploading the data as an <span class='extension'>h5ad</span> ([AnnData](https://anndata.readthedocs.io/en/latest/)) object.
    An AnnData object stores expression matrices and annotations efficiently while also providing convenient methods for accessing
    and storing intermediate analysis steps. Furthermore, an AnnData object can be highly compressed, thus saving precious upload (download) time.
    Below we provide a short Python snippet that can convert a <span class='extension'>csv</span> file into a maximally compressed AnnData object that
    can be later uploaded to Cellar. All you need is the `anndata` Python package that can be installed via `pip install anndata`. The following code
    reads a csv file `path/to/your/csv/file.csv` and writes it to another file `new/file/name.h5ad`.
    ```
    import anndata
    adata = anndata.read_csv('path/to/your/csv/file.csv')
    adata.write('new/file/name.h5ad', compression=9)
    ```
    Now upload `new/file/name.h5ad` to Cellar. This procedure may reduce a dataset's size from several GB to just a few MB.

    * There is a limit of 3GB for any uploaded dataset since large datasets take time
    to process. If you have a larger dataset, we recommend installing Cellar locally (follow instructions in the installation page),
    thus taking advantage of any computational resources you may have.

    * After the upload has finished, it will appears as the first item in the dropdown menu for datasets (see below) as `uploaded / datasetname`.

* **Dataset List**
    * This dropdown menu includes user uploaded datasets and server datasets. The server datasets are mainly provided by [HuBMAP](https://hubmapconsortium.org/). They are organized by data type (optional), data center and tissue.
    Cellar provides datasets from a total of five tissues: (lymph, spleen, thymus: scRNA-seq), (kidney: SNARE-seq), (heart: SCI-seq).

* **<span class='mbutton'>LOAD</span>**
    * This will load the selected dataset into memory which may
    take a few seconds depending on its size. 
    * Once a dataset is loaded, the number of samples (cells) and the number of features (genes) will be shown on the top left corner of the main body. If the dataset is annotated like most server datasets, then the main plot and analysis tabs will also get populated. Otherwise, you may wish to run (optionally preprocessing) dimensionality reduction (see below).
